### PR TITLE
Enable testing for Python 3.8

### PR DESCRIPTION
**Pull request by Jamim**
Hello,

Python 3.8 is [almost there](https://www.python.org/dev/peps/pep-0569/#schedule), so I believe that it would be nice to have tests for it on CI.

Best regards!

### CODE REVIEW

**Comment by yliaog** [comment]
3.8.0 final: Monday, 2019-10-21

we want to merge this after that, right?

**Comment by Jamim** [comment]
Hello @yliaog,
I believe it makes sense to merge it now in order to be aware in case something became broken for 3.8 due to some changes.

**Comment by micw523** [comment]
/hold
Even though Python 3.8 will be here soon, Travis CI will probably be ready much later. Therefore there’s no rush getting this PR in since we’ll have to wait.

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +28,10 @@ matrix:
        env: TOXENV=py37
      - python: 3.7
        env: TOXENV=py37-functional
+     - python: 3.8-dev
```
Comment (micw523) [inline .travis.yml]: 3.8-dev is still a development build and is not ready for production yet.

**Comment by Jamim** [comment]
In fact, Travis CI is ready.
Please take a look at [the corresponding build](https://travis-ci.org/kubernetes-client/python/builds/589666045).

File: .travis.yml (hunk 1 above)
Comment (Jamim) [reply .travis.yml]: But it's good enough for testing purposes.

**Comment by micw523** [comment]
Yet, 3.8 dev is not 3.8 final. Also, I don’t know how I selected approve for this PR - I meant “comment”.

File: .travis.yml (hunk 1 above)
Comment (micw523) [reply .travis.yml]: The reason we want to test a certain version is that we’ll be confident supporting this version in PyPI wheels. Therefore, I would stick to stable release only.

**Comment by Jamim** [comment]
https://pythoninsider.blogspot.com/2019/08/python-380b4-is-now-available-for.html
> We `strongly encourage` maintainers of third-party Python projects to `test with 3.8` during the beta phase and report issues found to [the Python bug tracker](https://bugs.python.org) as soon as possible. Please note this is the last beta release, there is not much time left to identify and fix issues before the release of 3.8.0. If you were hesitating trying it out before, `now is the time`.

**Comment by micw523** [comment]
https://pythoninsider.blogspot.com/2019/08/python-380b4-is-now-available-for.html

> Please keep in mind that this is a preview release and its use is not recommended for production environments. 

Please do notice that this repo is used in production environments. We won't be confident in supporting Python 3.8 unless it's production ready.

**Comment by micw523** [comment]
/hold cancel
Please update the environment to 3.8 stable and we'll see from there

**Comment by Jamim** [comment]
Hello @micw523,
`3.8` is [not available](https://travis-ci.org/Jamim/kubernetes-client-python/jobs/597896582) on Travis CI yet.
Currently, there is `3.8-dev` only.

**Comment by Jamim** [comment]
Hi @micw523,
I believe this PR is ready to be merged now.

**Comment by micw523** [comment]
/lgtm
/cc @roycaihw 
/assign @roycaihw 
This looks good to me.

File: setup.py [hunk 2]
```diff
@@ -77,5 +77,6 @@
         "Programming Language :: Python :: 3.5",
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
```
Comment (micw523) [inline setup.py]: As a side note, we'll need py38 wheels in the next release possibly.

**Pull request merged**